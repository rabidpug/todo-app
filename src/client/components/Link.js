// @flow

import { Icon, } from 'antd';
import React from 'react';

type Props = {
  label: string,
  icon: string,
  key: string,
  onClick: Function,
};
const Link: Function = ( {
  label, icon, onClick,
}: Props ) => (
  <div
    onClick={ onClick }
    style={ { width: '100%', } }>
    <Icon type={ icon } />
    {label}
  </div>
);

export default Link;
