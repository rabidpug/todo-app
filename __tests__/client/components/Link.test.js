import Link from 'Components/Link';
import React from 'react';
import { shallow, } from 'enzyme';
import { shallowToJson, } from 'enzyme-to-json';

describe(
  'Link', () => {
    it(
      'Should render the Link', () => {
        const wrapper = shallow( <Link /> );

        expect( shallowToJson( wrapper ) ).toMatchSnapshot();
      }
    );
  }
);
