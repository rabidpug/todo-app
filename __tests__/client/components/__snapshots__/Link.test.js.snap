// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Should render the Link 1`] = `
<div
  style={
    Object {
      "width": "100%",
    }
  }
>
  <Icon />
</div>
`;
